src/org/omegat/gui/main/ProjectUICommands.java

```diff
@@ -537,19 +551,49 @@ protected Void doInBackground() throws Exception {
                     }
 
                     final ProjectProperties propsP = props;
-                    final File restoreOnFailFinal = restoreOnFail;
+                    final File rewriteOnSuccessFinal = rewriteOnSuccess;
+                    final boolean needToSavePropertiesFinal = needToSaveProperties;
+                    // We have to extract the processing related to needToSaveProperties{,Final}
+                    // out of  the code block of  Core.executeExclusively()
+                    // and place it here, so that we can change the value of needToSaveProperties
+                    // in our update scheme for teamwork setting.
+                    // Sanity check of possibly modified new project property.
+                    final boolean succeeded = ProjectFactory.loadProject(propsP, true);
+                    if (rewriteOnSuccessFinal != null) { // teamwork case.
+                        if (succeeded) {    // remote content with minor mods seems sane.
+                            if (needToSavePropertiesFinal) {
+                                Log.logWarningRB("REMOTE_PROJECT_CONTENT_WAS_MODIFIED_TO_ADD_MAPPING_SO_NOT_USED");
+                                // Do not save the content of the remote file because we missed mapping.
+                                // So in our new scheme of backup and updating, we don't need to save the file
+                                // after this code block any more.
+                                needToSaveProperties = false;
+                                Files.deleteIfExists(rewriteOnSuccessFinal.toPath()); // remove "omegat.project.NEW"
+                            }
+                        }
+                    }
+                    // Renaming is done inside a critical region using Core.executeExclusively().
                     Core.executeExclusively(true, () -> {
-                        boolean succeeded = ProjectFactory.loadProject(propsP, true);
-                        if (restoreOnFailFinal != null) {
-                            if (succeeded) {
-                                Files.deleteIfExists(restoreOnFailFinal.toPath());
-                            } else {
-                                Files.move(restoreOnFailFinal.toPath(), projectFile.toPath(),
-                                        StandardCopyOption.REPLACE_EXISTING);
+                        if (rewriteOnSuccessFinal != null) { // teamwork case.
+                            if (succeeded) {    // remote content with minor mods seems sane.
+                                if (needToSavePropertiesFinal) {
+                                    // This part is moved out of this lambda and put in the preceding position because
+                                    // we need to modify needToSaveProperties to false.
+                                } else {
+                                    Log.logWarningRB("REMOTE_PROJECT_CONTENT_OVERRIDES_THE_CURRENT_PROJECT");
```

Comment:
No need to explain the patch in a comment. Please invert if logic.
```suggestion
                                if (!needToSavePropertiesFinal) {
                                    Log.logWarningRB("REMOTE_PROJECT_CONTENT_OVERRIDES_THE_CURRENT_PROJECT");
```